src/blockchain/deposit_strategy/deposit_transaction_sender.py

```diff
@@ -16,9 +23,16 @@ class Sender:
     Chain senders for deposit transactions.
     """
     _TIMEOUT_IN_BLOCKS = 6
+    _next_sender: Sender = None
 
-    def __init__(self, w3: Web3):
+    def __init__(self, w3: Web3, gas_price_calculator: GasPriceCalculator, strategy: BaseDepositStrategy):
         self._w3 = w3
+        self._gas_price_calculator = gas_price_calculator
+        self._strategy = strategy
+
+    def add_sender(self, sender: Sender):
```

Comment:
Linked list is not obvious (with recursion), better use just list
But I propose we can do it with one if only. Not sure if there will be another type of sender

Reply:
Removed